### PR TITLE
Add support for differentiating while loops in forward mode

### PR DESCRIPTION
**Pull request by parth-07**
This PR adds support for differentiating `while` and `do-while` loops in forward differentiation mode.

Progress checklist:
- [x] Add support for differentiating `while` and `do-while` loops in forward mode
- [x] Add tests

### CODE REVIEW

File: lib/Differentiator/ForwardModeVisitor.cpp [hunk 1]
```diff
@@ -1018,4 +1018,120 @@ namespace clad {
...
+      CompoundStmt* Block = endBlock();
+      endScope();
+      bodyResult = Block;
+    }
+
+    DoStmt* DoStmtDiff =
+        m_Sema
+            .ActOnDoStmt(/*DoLoc=*/noLoc, bodyResult, /*WhileLoc=*/noLoc,
+                         /*CondLParen=*/noLoc, clonedCond, /*CondRParen=*/noLoc)
+            .getAs<DoStmt>();
```
Comment (vgvassilev) [inline lib/Differentiator/ForwardModeVisitor.cpp:1131]:
```suggestion
    Stmt* S =
        m_Sema
            .ActOnDoStmt(/*DoLoc=*/noLoc, bodyResult, /*WhileLoc=*/noLoc,
                         /*CondLParen=*/noLoc, clonedCond, /*CondRParen=*/noLoc)
            .get();
```
We do not need the concrete type of the statement for the return result of the function.

**Review by vgvassilev — COMMENTED** [review]
Please squash the commits of this PR.

File: lib/Differentiator/ForwardModeVisitor.cpp (hunk 1 above)
Comment (parth-07) [reply lib/Differentiator/ForwardModeVisitor.cpp:1131]: Okay

File: lib/Differentiator/ForwardModeVisitor.cpp [hunk 2]
```diff
@@ -1018,4 +1018,120 @@ namespace clad {
         IntegerLiteral::Create(m_Context, zero, m_Context.IntTy, noLoc);
     return StmtDiff(Clone(BL), constant0);
   }
+
+  /// Creates and returns a compound statement having statements as follow:
+  /// {`stmt`, all the statement of `CS` in sequence}
+  static CompoundStmt* InsertAtFrontOfCompoundStmt(ASTContext& C,
```
[the remark this answers is not in the document]
Comment (parth-07) [reply lib/Differentiator/ForwardModeVisitor.cpp:1024]: Yes, the suggested name captures the intent in a much better way.

**Review by vgvassilev — APPROVED** [review]
Lgtm!

**Pull request merged**